knora/dsplib/utils/xml_upload.py

```diff
@@ -910,94 +918,340 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
+
+    Returns:
+        nonapplied_xml_texts: the xml texts that could not be updated
+    """
+
+    print('Update the stashed XML texts...')
+    for resource, link_props in stashed_xml_texts.copy().items():
         print(f'Update XML text(s) of resource "{resource.id}"...')
-        res_iri = res_iri_lookup[resource.id]
-        try:
-            update_xml_texts(
-                resource=resource,
-                res_iri=res_iri,
-                link_props=link_props,
-                res_iri_lookup=res_iri_lookup,
-                con=con,
-                verbose=verbose
-            )
-        except BaseError as err:
-            print(f'BaseError while updating an XML text of resource "{resource.id}": {err.message}')
-            continue
-        except Exception as exception:
-            print(f'Exception while updating an XML text of resource "{resource.id}": {exception}')
-            continue
+        res_iri = id2iri_mapping[resource.id]
+        existing_resource = con.get(path=f'/v2/resources/{quote_plus(res_iri)}')
+        for link_prop, hash_to_value in link_props.items():
+            existing_values = existing_resource[link_prop.name]
+            if not isinstance(existing_values, list):
+                existing_values = [existing_values, ]
+            for existing_value in existing_values:
+                old_xmltext = existing_value.get("knora-api:textValueAsXml")
+                if not old_xmltext:
+                    continue
+
+                # strip all xml tags from the old xmltext, so that the pure text itself remains
+                pure_text = re.sub(r'(<\?xml.+>\s*)?<text>\s*(.+)\s*<\/text>', r'\2', old_xmltext)
+
+                # if the pure text is a hash, the replacement must be made
+                if pure_text not in hash_to_value:
+                    continue
+                new_xmltext = hash_to_value[pure_text]
+
+                # replace the outdated internal ids by their IRI
+                for _id, _iri in id2iri_mapping.items():
+                    new_xmltext.regex_replace(f'href="IRI:{_id}:IRI"', f'href="{_iri}"')
+
+                # prepare API call
+                jsonobj = {
+                    "@id": res_iri,
+                    "@type": resource.restype,
+                    link_prop.name: {
+                        "@id": existing_value['@id'],
+                        "@type": "knora-api:TextValue",
+                        "knora-api:textValueAsXml": new_xmltext,
+                        "knora-api:textValueHasMapping": {
+                            '@id': 'http://rdfh.ch/standoff/mappings/StandardMapping'
+                        }
+                    },
+                    "@context": existing_resource['@context']
+                }
+                jsondata = json.dumps(jsonobj, indent=4, separators=(',', ': '), cls=KnoraStandoffXmlEncoder)
+
+                # execute API call
+                answer = None
```

Comment:
I would call this `response` instead of `answer`.

Reply:
yes